### PR TITLE
Upgrade to Treelite 2.1.0

### PR DESCRIPTION
**Pull request by hcho3**
Required by rapidsai/cuml#4220. See the CI result in rapidsai/cuml#4220 which is testing cuML with Treelite 2.1.0.

### CODE REVIEW

**Comment by Ethyling** [comment]
https://github.com/rapidsai/cuml/pull/4220 CI is failing due to `mamba remove --force` issue. Waiting for CI to pass before merging

**Review by Ethyling — APPROVED** [review]

**Comment by ajschmidt8** [comment]
CI is confirmed passing on https://github.com/rapidsai/cuml/pull/4220. This PR can be merged.

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**
